packages/dx-scheduler-core/src/plugins/current-time-indicator/helpers.tsx

```diff
@@ -0,0 +1,40 @@
+import { PureComputed } from '@devexpress/dx-core';
+import moment from 'moment';
+import {
+  IsCellShadedFn, IsShadedAppointment,
+  GetCurrentTimeIndicatorTopFn,
+} from '../../types';
+
+export const isMonthCell: PureComputed<
+  [boolean | undefined], boolean
+> = otherMonth => otherMonth !== undefined;
+
+export const isShadedAppointment: IsShadedAppointment = (
+  { data: appointmentData }, currentTime, shadePastAppointments,
+) => {
+  const momentCurrentDate = moment(currentTime);
+  if (appointmentData.allDay) {
+    return momentCurrentDate.isAfter(appointmentData.endDate as Date, 'day')
+    && shadePastAppointments;
+  }
+  if (momentCurrentDate.isAfter(appointmentData.endDate as Date)) {
+    return shadePastAppointments;
+  }
+  return false;
```

Comment:
Maybe
```jsx
if (shadePastAppointments) {
  const allDayAppointmentAfter = appointmentData.allDay
    && momentCurrentDate.isAfter(appointmentData.endDate as Date, 'day');
  const appointmentAfter = momentCurrentDate.isAfter(appointmentData.endDate as Date);
  return allDayAppointmentAfter  || appointmentAfter;
}
return false
```

Reply:
This will reduce count of `return` and `if`

Reply:
This won't help. We must differentiate between allday appointments and non-allday appointments. And if we use this:
```
return allDayAppointmentAfter  || appointmentAfter;
```
In this case we do not differentiate because if `appointmentAfter` is `true` then the whole expression is true. Despite the fact that the appointment may last for a whole day